olmo/checkpoint.py

```diff
@@ -516,6 +516,7 @@ def restore_checkpoint(
         *,
         local_cache: Optional[PathOrStr] = None,
         load_optimizer_state: bool = True,
+        load_trainer_state: bool = True,
     ) -> Dict[str, Any]:
```

Comment:
Looks like the intended return type is now `Optional[Dict[str, Any]]`?
```suggestion
    ) -> Optional[Dict[str, Any]]:
```